pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -0,0 +1,255 @@
+			},
+		},
+
+		gvr("extensions"): &artifact{
+			newObject: func() client.Object { return new(extensionsv1alpha1.Extension) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				return validation.ValidateExtension(n.(*extensionsv1alpha1.Extension))
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				return validation.ValidateExtensionUpdate(n.(*extensionsv1alpha1.Extension), o.(*extensionsv1alpha1.Extension))
+			},
+		},
+
+		gvr("infrastructures"): &artifact{
+			newObject: func() client.Object { return new(extensionsv1alpha1.Infrastructure) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				return validation.ValidateInfrastructure(n.(*extensionsv1alpha1.Infrastructure))
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				return validation.ValidateInfrastructureUpdate(n.(*extensionsv1alpha1.Infrastructure), o.(*extensionsv1alpha1.Infrastructure))
+			},
+		},
+
+		gvr("networks"): &artifact{
+			newObject: func() client.Object { return new(extensionsv1alpha1.Network) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				return validation.ValidateNetwork(n.(*extensionsv1alpha1.Network))
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				return validation.ValidateNetworkUpdate(n.(*extensionsv1alpha1.Network), o.(*extensionsv1alpha1.Network))
+			},
+		},
+
+		gvr("operatingsystemconfigs"): &artifact{
+			newObject: func() client.Object { return new(extensionsv1alpha1.OperatingSystemConfig) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				return validation.ValidateOperatingSystemConfig(n.(*extensionsv1alpha1.OperatingSystemConfig))
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				return validation.ValidateOperatingSystemConfigUpdate(n.(*extensionsv1alpha1.OperatingSystemConfig), o.(*extensionsv1alpha1.OperatingSystemConfig))
+			},
+		},
+
+		gvr("workers"): &artifact{
+			newObject: func() client.Object { return new(extensionsv1alpha1.Worker) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				return validation.ValidateWorker(n.(*extensionsv1alpha1.Worker))
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				return validation.ValidateWorkerUpdate(n.(*extensionsv1alpha1.Worker), o.(*extensionsv1alpha1.Worker))
+			},
+		},
+	}
+
+	h := handler{
+		logger:    logger,
+		artifacts: artifacts,
+	}
+
+	return &h
+}
+
+type handler struct {
+	decoder   *admission.Decoder
+	logger    logr.Logger
+	artifacts map[metav1.GroupVersionResource]*artifact
+}
+
+var _ admission.Handler = &handler{}
+
+func (h *handler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *handler) Handle(ctx context.Context, ar admission.Request) admission.Response {
+	h.logger.Info("Validating resource", "type", ar.Resource, "operation", ar.Operation)
```

Comment:
This log message here and the one below at line 226 are redundant with each other. Please leave only one (I think rather the second one), and include all relevant details - resource (or kind), operation, and full resource name.